src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -266,6 +266,7 @@ func filterNaNSeries(
 func renderResultsJSON(
 	w io.Writer,
 	series []*ts.Series,
+	exemplarsList ts.ExemplarsList,
```

Comment:
Are you doing anything with this? Also can we give it a separate render method so that it doesn't leak into regular results?

Reply:
I'd recommend not downsampling here; rather doing it where points get downsampled or merged; probably in FetchResultToBlockResult for initial load, then in mutating functions themselves.

Reply:
In that case, keep exemplar on a per-series basis?